contracts/stabilizer/PSMRouter.sol

```diff
@@ -32,15 +69,61 @@ contract PSMRouter is IPSMRouter {
     /// @dev This wraps ETH and then calls into the PSM to mint the fei. We return the amount of fei minted.
     /// @param to The address to mint fei to
     /// @param minAmountOut The minimum amount of fei to mint
-    /// @param deadline The minimum amount of fei to mint
+    /// @param deadline The deadline for this order to be filled
     function mint(address to, uint256 minAmountOut, uint256 deadline) external payable ensure(deadline) returns (uint256) {
         return _mint(to, minAmountOut);
     }
 
+    /// @notice Redeems fei for ETH
+    /// First pull user FEI into this contract
+    /// Then call redeem on the PSM to turn the FEI into weth
+    /// Withdraw all weth to eth in the router
+    /// Send the eth to the specified recipient
+    /// @param to the address to receive the eth
+    /// @param amountFeiIn the amount of FEI to redeem
+    /// @param minAmountOut the minimum amount of weth to receive
+    function redeem(address to, uint256 amountFeiIn, uint256 minAmountOut) external override returns (uint256) {
+        return _redeem(to, amountFeiIn, minAmountOut);
+    }
+
+    /// @notice Redeems fei for ETH
+    /// First pull user FEI into this contract
+    /// Then call redeem on the PSM to turn the FEI into weth
+    /// Withdraw all weth to eth in the router
+    /// Send the eth to the specified recipient
+    /// @param to the address to receive the eth
+    /// @param amountFeiIn the amount of FEI to redeem
+    /// @param minAmountOut the minimum amount of weth to receive
+    /// @param deadline The deadline for this order to be filled
+    function redeem(address to, uint256 amountFeiIn, uint256 minAmountOut, uint256 deadline) external ensure(deadline) returns (uint256) {
+        return _redeem(to, amountFeiIn, minAmountOut);
+    }
+
+    /// @notice function to receive ether from the weth contract when the redeem function is called
+    /// will not accept eth unless there is an active redemption.
+    fallback() external payable {
+        require(redeemActive, "PSMRouter: redeem not active");
```

Comment:
receive() external payable {
        assert(msg.sender == WETH); // only accept ETH via fallback from the WETH contract
    }

This code pattern used in Uniswap V2 router is more gas efficient and accurate

Reply:
Accepted that change but used require as I like to see error strings on failure.